### DISCUSSION TITLE
Many CHILD SAs for one IKE SA

### DISCUSSION THREAD

**Discussion by misch42** [strongswan/strongswan#1840] [General]
Hi,

I set up some VPNs on a gateway. For some tunnels our strongswan ends up with many CHILD SAs for the IKE SA. This might be up to 200 CHILD SAs, which is definitely too much.

When there is a Cisco on the other side we receive new CREATE_CHILD_SA about every 30 seconds.

Cisco says, it just reminds every 30 seconds about rekeying, which is, IMHO bullshit. 

Any ideas?

**Comment by misch42** [comment]
when there is a sophos on the other side we see a lot of:

Aug 08 13:45:46 vpn-gw charon[720]: 13[IKE] establishing CHILD_SA partner-01-net{97207} reqid 13
Aug 08 13:45:46 vpn-gw charon[720]: 13[IKE] establishing CHILD_SA partner-01-net{97207} reqid 13
Aug 08 13:45:46 vpn-gw charon[720]: 13[ENC] generating CREATE_CHILD_SA request 2400 [ N(REKEY_SA) SA No KE TSi TSr ]
Aug 08 13:45:46 vpn-gw charon[720]: 13[NET] sending packet: from x.x.x.x[500] to y.y.y.y[500] (496 bytes)
Aug 08 13:45:46 vpn-gw charon[720]: 14[NET] received packet: from y.y.y.y[500] to x.x.x.x[500] (464 bytes)
Aug 08 13:45:46 vpn-gw charon[720]: 14[ENC] parsed CREATE_CHILD_SA response 2400 [ SA No KE TSi TSr ]
Aug 08 13:45:46 vpn-gw charon[720]: 14[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ
Aug 08 13:45:46 vpn-gw charon[720]: 14[CFG] unable to install policy X.X.X.X/32 === Y.Y.Y.Y/24 in for reqid 13, the same policy for reqid 8 exists
Aug 08 13:45:46 vpn-gw charon[720]: 14[CFG] unable to install policy X.X.X.X/32 === Y.Y.Y.Y/24 fwd for reqid 13, the same policy for reqid 8 exists
Aug 08 13:45:46 vpn-gw charon[720]: 14[IKE] unable to install IPsec policies (SPD) in kernel
(...)
Aug 08 13:45:46 vpn-gw charon[720]: 08[IKE] CHILD_SA rekeying failed, trying again in 11 seconds

**Reply by tobiasbrunner** [reply]
Check the traffic selectors for the established CHILD_SAs, there might be some conflicts there.

**Comment by misch42** [comment]
In both cases (Cisco and sophos) we end up with 100 - 200 CHILD SAs for every IKE SA.

Any ideas?

**Comment by tobiasbrunner** [comment]
> Cisco says, it just reminds every 30 seconds about rekeying,

What does "reminds ... about rekeying" mean if it actually does a rekeying (i.e. creates a new CHILD_SA)?

Are these proper rekeyings? With IKEv2, the CREATE_CHILD_SA request would contain a REKEY_SA notify that refers to the old SA.

Also, does it not delete the replaced SAs? strongSwan obviously won't remove them until they expire (i.e. to get rid of them you'd have to reduce the lifetime configured in strongSwan as well).
